src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs

```diff
@@ -141,6 +141,13 @@ internal override void ProcessResponse(HttpResponseMessage response)
             }
             else if (ShouldSaveToOutFile)
             {
+                // Check if OutFile is a folder
+                if (Directory.Exists(QualifiedOutFile))
+                {
+                    // Get file name from last segment of Uri
+                    OutFile = Path.Combine(OutFile, System.Net.WebUtility.UrlDecode(response.RequestMessage.RequestUri.Segments[^1]));
```

Comment:
Now we do path resolving twice since QualifiedOutFile is not cache result.
I'd expect we cache QualifiedOutFile in local variable, modify it here and use below.

Reply:
And we need tests for the new scenarios.